src/app/beer_garden/db/mongo/models.py

```diff
@@ -474,12 +476,24 @@ def clean(self):
                 f"consistent with has_parent value of {self.has_parent}"
             )
 
+        try:
```

Comment:
I'm going to guess that the problem you ran into with trying to use `_changed_fields` is that it doesn't get set (or even exist) if we're dealing with a new object.  You already have the default value getting set for `status_updated_at` on new Requests, so we shouldn't have to worry about that case.  Which means I think you can just something like this:

```python
# In the MongoModel base class that Request builds off of:
@property
def changed_fields(self):
    return getattr(self, "_changed_fields", [])

# And then in place of this try / except block:
if "status" in self.changed_fields:                                                                                                                                                                       
   self.status_updated_at = datetime.datetime.utcnow()
```

That allows us to avoid the extra database lookup that you've got in there now.

Reply:
Changes are made. Had to force-push a new commit, since I learned the hard way that `git pull --rebase` does not do what I thought it did. 

There is some strange behavior, however - see the unit tests. If the first `save()` is removed from the second unit test, where the `status` field is actually changed, the unit test fails. `_changed_fields` seems to be accurate after the first save, however.

Reply:
I think this is the expected behavior based on the way the fixture is setup.  The fixture doesn't actually call `save()`, so in the test you're dealing with model instance that hasn't actually been saved to the DB yet.  That means the first time you call `save()` in the test, you're going down the path where `_created` is `True` and `_changed_fields` is not set.  Basically, without that first save, you're not really changing the `status` field, you're setting its initial value.

All that is to say, give the fixture you're working with, your test is doing the right thing.